e2e/credentials-server.e2e.spec.ts

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * This file includes end-to-end testing for the HTTP control interface
+ */
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { spawnSync } from 'child_process';
+
+import { expect, test } from '@playwright/test';
+import { BrowserContext, ElectronApplication, Page, _electron } from 'playwright';
+
+import fetch from 'node-fetch';
+import { createDefaultSettings, playwrightReportAssets } from './utils/TestUtils';
+import paths from '@/utils/paths';
+import { ServerState } from '@/main/commandServer/httpCommandServer';
+import { wslHostIPv4Address } from '@/utils/networks';
+import { spawnFile } from '@/utils/childProcess';
+import { findHomeDir } from '@/config/findHomeDir';
+
+let credStore: string;
+
+function haveCredentialServerHelper(): boolean {
+  // Not using the code from `httpCredentialServer.ts` because we can't use async code at top-level here.
+  const dockerConfigPath = path.join(findHomeDir() ?? '', '.docker', 'config.json');
+
+  try {
+    const contents = JSON.parse(fs.readFileSync(dockerConfigPath).toString());
+    const credStoreAttempt = contents.credsStore;
+
+    if (!credStoreAttempt) {
+      return false;
+    }
+    credStore = credStoreAttempt;
+    const result = spawnSync(`docker-credential-${ credStore }`, { input: 'list', stdio: 'pipe' });
+
+    return !result.error;
+  } catch (err: any) {
+    return false;
```

Comment:
This means we will not run the tests if:
- There's no `~/.docker/config.json` (e.g. fresh VM)
- There's no `credStore` set (shouldn't we fall back?)

Reply:
Well, I suppose I could write a mock `docker-credential-mock` executable that implements enough of the credential helper API that the tests in this file pass, but that doesn't seem to me to be a worthwhile use of time.

Better to understand that this is a useful test for current manual testing, and when we start doing bats-based full-system integration testing, it will have a use after the docker CLI is installed, and the bats test can set up `~/.docker/config.json` with a `credsStore` field.

But for now I think this is a reasonable limitation.